offer.py

```diff
@@ -1,59 +1,60 @@
 from abc import ABC, abstractmethod
-from uuid import uuid4, UUID
-from decimal import Decimal
 from datetime import datetime
+from decimal import Decimal
+from typing import List
+from uuid import uuid4, UUID
 
-from manager import Manager, singleton
+from car import Car, CarType
+from payment import PaymentHandler, PaymentHandlerType, create_handler, CashPayHandler
+from user import UserManager
+from passenger import Passenger
+from dataclasses import dataclass
 from driver import Driver
+from manager import Manager, singleton
 from map import Location, Map
-from client import ClientManager
-from customer import Customer
-from car import Car, CarType
 
 
-# TODO: dataclass?
+@dataclass
 class Offer:
-    def __init__(self, customer: Customer, offer_time: datetime, departure_point: Location,
-                 destination_point: Location, car_type, price: Decimal):
-        self.customer_id = customer.id
-        self.offer_time = offer_time
-        self.departure_point = departure_point
-        self.destination_point = destination_point
-        self.car_type = car_type
-        self.price = price
+    car_type: CarType = CarType.ECONOMY
+    departure_point: Location = Location(0, 0)
+    destination_point: Location = Location(0, 0)
+    offer_time: datetime = datetime(year=0, month=0, day=0)
+    passenger: Passenger = Passenger('admin', 'password')
+    payment_handler: PaymentHandler = CashPayHandler(passenger)
+    price: Decimal = Decimal(0.0)
```

Comment:
В датаклассах не обязательно писать значения по умолчанию. Особенно в данном случае, иначе можно будет создать объект со значениями по умолчанию, а лучше всё-таки создавать объекты со значащими переменными

```suggestion
    departure_point: Location
    destination_point: Location
    offer_time: datetime
    passenger: Passenger
    payment_handler: PaymentHandler
    price: Decimal
```

Reply:
Мы это сделали, чтобы mypy не ругался на Optional[X] типы, чтобы не делать лишние проверки на None

Потому что есть гарантия, что Offer создается только при помощи билдера, соответственно все поля заполняются в процессе и точно не None

Reply:
С точки зрения кодинга, гарантий, что Offer создается только при помощи билдера, нет. Я могу в любом месте создать объект класса Offer. Поэтому лучше сделать проверки на None

Reply:
Хорошо, но где вставлять эти проверки? Есть вариант сделать так, чтобы все функции принимали Optional, но тогда придется писать проверку внутри каждой функции, причем все проверки будут выглядеть одинаково (если None, то raise что-нибудь). Есть ли более красивое решение? Можно оставить сигнатуры функций прежними, но уже на пользователя скинуть ответственность за то, что он не кинет в них None - это плохой вариант, который генерирует множество проверок уже со стороны пользователя

Reply:
Давайте завтра на паре обсудим